src/Powercord/apis/settings/index.js

```diff
@@ -0,0 +1,157 @@
+const { randomBytes, scryptSync, createCipheriv, createDecipheriv } = require('crypto');
+const { WEBSITE } = require('powercord/constants');
+const { getComponentByDisplayName, React } = require('powercord/webpack');
+const { get, post } = require('powercord/http');
+const { API } = require('powercord/entities');
+
+const Category = require('./category');
+
+module.exports = class Settings extends API {
+  constructor () {
+    super();
+
+    this.settings = {};
+    this.tabs = [];
+  }
+
+  // Classic stuff
+  async apiDidLoad () {
+    await this.download();
+    this._interval = setInterval(this.upload.bind(this), 10 * 60 * 1000);
+  }
+
+  async apiWillUnload () {
+    clearInterval(this._interval);
+    await this.upload();
+  }
+
+  // Categories
+  registerTab (section, displayName, render) {
+    if (!section.match(/^[a-z0-9_-]+$/i)) {
+      return this.error(`Tried to register a settings panel with an invalid ID! You can only use letters, numbers, dashes and underscores. (ID: ${section})`);
+    }
+
+    if (this.tabs.find(s => s.section === section)) {
+      return this.error(`Key ${section} is already used by another plugin!`);
+    }
+
+    this.tabs.push({
+      section,
+      label: displayName,
+      element: this._renderSettingsPanel.bind(this, displayName, render)
+    });
+  }
+
+  unregisterTab (section) {
+    this.tabs = this.tabs.filter(s => s.section !== section);
+  }
+
+  // Manage settings
+  getCategory (category) {
+    this._ensureCategory(category);
+    return this.settings[category];
+  }
+
+  get (category, setting, defaultValue) {
+    this._ensureCategory(category);
+    return this.settings[category].get(setting, defaultValue);
+  }
+
+  set (category, setting, value) {
+    this._ensureCategory(category);
+    this.settings[category].set(setting, value);
+  }
+
+  _ensureCategory (category) {
+    if (!this.settings[category]) {
+      this.settings[category] = new Category(category);
+      this.settings[category]._load();
+    }
+  }
+
+  _renderSettingsPanel (title, contents) {
+    let panelContents;
+    try {
+      panelContents = React.createElement(contents);
+    } catch (e) {
+      this.error('Failed to render settings panel, check if your function returns a valid React component!');
+      panelContents = null;
+    }
+
+    const h2 = React.createElement(getComponentByDisplayName('FormTitle'), { tag: 'h2' }, title);
+    return React.createElement(getComponentByDisplayName('FormSection'), {}, h2, panelContents);
```

Comment:
is this safe to be not awaited?

Reply:
getComponentByDisplayName is an helper to get an AsyncComponent instance that will await the module by itself. No need to await it